source/PlayerInfo.cpp

```diff
@@ -795,6 +815,45 @@ const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 
 
 
+// Set the flagship (on departure or during flight).
+void PlayerInfo::SetFlagship(Ship &other)
+{
+	// Remove active data in the old flagship.
+	if(flagship && flagship.get() != &other)
```

Comment:
Can't we return early if the flagship is already `other`? Or guard the call to MoveFlagshipBegin and SetFlagship.

Reply:
No, it is possible that some on-planet actions already resulted in `flagship.get() == &other` being true, while some of the other actions were done yet. (The on-planet flagship handling only looks at positions in the ships list.)

Also, the player could have unparked some ships just before launch, those ships also need to be updated on line 831.